File: typescript/behavioral_pattern/src/template_method/HotChocolate.ts
import Beverage from "./Beverage";

class HotChocolate extends Beverage {
    protected brew(): void {
        console.log("Mixing chocolate powder...");
    }

    protected addCondiments(): void {
        console.log("Adding whipped cream and marshmallows...");
    }

    protected customerWantsCondiments(): boolean {
        return true;
    }
}

export default HotChocolate;

File: behavioral_pattern/template_method/HotChocolate.java
package template_method;

public class HotChocolate extends Beverage {
    @Override
    protected void brew() {
        System.out.println("Mixing chocolate powder...");
    }

    @Override
    public void addCondiments() {
        System.out.println("Adding whipped cream and marshmallows...");
    }

    @Override
    protected boolean customerWantsCondiments() {
        return true;
    }
}
